### ISSUE TITLE
Release hotfix_fake4 for MSBUILD VS2019 Support

### ISSUE THREAD

**Issue by embix** [fsharp/FAKE#2337]
### Description

Building with (only) VS2019 installed using most recent FAKE4 (4.64.14) fails.

### Repro steps

Please provide the steps required to reproduce the problem

1. Delete all VS Versions older than VS2019 (if any).

2. Run any build.fsx of any (otherwise working) FSharp project.

### Expected behavior

Build simply works (using any recent MSBUILD and FSharp Compiler stuff)

### Actual behavior

Falling back to ancient version of MSBUILD, providing help about what went wrong (There is no Microsoft.Build.Utilities.Core, Version=14.0.0.0 on nuget, oldest version there is would be 14.3)

```
Building project: Company.Topic.sln
  c:\Windows\Microsoft.NET\Framework\v4.0.30319\MSBuild.exe  Company.Topic.sln /t:Build /m  /nodeReuse:False  /v:q  /p:RestorePackages="False" /p:Configuration="Release" /p:FSharpTargetsPath="C:\git\Topic\packages\build\FSharp.Compiler.Tools\tools\Microsoft.FSharp.Targets" /logger:Fake.MsBuildLogger+ErrorLogger,"C:\git\Topic\packages\build\FAKE\tools\FakeLib.dll"

Microsoft (R) Build Engine version 4.8.3761.0
[Microsoft .NET Framework, version 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.

C:\git\Topic\packages\build\FSharp.Compiler.Tools\tools\Microsoft.FSharp.Targets(183,9): error MSB4062: The "FSharpEmbedR
esXSource" task could not be loaded from the assembly C:\git\Topic\packages\build\FSharp.Compiler.Tools\tools\FSharp.Buil
d.dll. Could not load file or assembly 'Microsoft.Build.Utilities.Core, Version=14.0.0.0, Culture=neutral, PublicKeyTok
en=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask
> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public
 class that implements Microsoft.Build.Framework.ITask. [C:\git\Topic\Topic.Domain\Topic.Domain.fsproj]

```

### Known workarounds

Build directly in VS2019.

### Related information 

* Windows 10, 1809 (Build 17763.557)
* Branch
* .NET Runtime
* Breaks the build -> severe
* Version of FAKE (4.64.14 - Why is there no related tag in the public repo?!)

Relevant Code from tag 4.64.13:
```
let knownMsBuildEntries =
    [
        { Version = "15.0"; Paths = [@"\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin"
                                     @"\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin"
                                     @"\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin"
                                     @"\MSBuild\15.0\Bin"
                                     @"\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin"] }
        { Version = "14.0"; Paths = [@"\MSBuild\14.0\Bin"] }
        { Version = "12.0"; Paths = [@"\MSBuild\12.0\Bin"; @"\MSBuild\12.0\Bin\amd64"] }
    ]

let oldMsBuildLocations =
    [ @"c:\Windows\Microsoft.NET\Framework\v4.0.30319\";
      @"c:\Windows\Microsoft.NET\Framework\v4.0.30128\";
      @"c:\Windows\Microsoft.NET\Framework\v3.5\"
    ]
```

Fixed code from hotfix_fake4 (31a93745b427da20308ee8c6f90710b57596956d):
```
 let knownMsBuildEntries =
     [
        { Version = "16.0"; Paths = [@"\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin"
                                     @"\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin"
                                     @"\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin"
                                     @"\MSBuild\Current\Bin"
                                     @"\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin"] }
```

TL;DR: As there already is a fixed version (hotfix_fake4, currently pointing to 31a93745b427da20308ee8c6f90710b57596956d), there's no need for a Pull-Request. *All we need is a release for FAKE4.*

**Comment by matthid** [comment]
Is there a particular reason why you upgrade your infrastructure but expect legacy fake version to still work? Considering that for example [all FCT usage was broken](https://github.com/dotnet/fsharp/issues/6689) you most likely had to do changes to get VS2017 and VS2019 working anyway.

While I agree that this should probably be fixed I guess we should slowly phase out FAKE 4 support.
At the very least we can expect people to use the 5.X nuget package at this time, which shouldn't contain breaking changes. Does it work if you upgrade to latest?

**Comment by embix** [comment]
I just cloned a project and wondered why it didn't compile on my machine whereas it builds in CI and on my coworkers machines. I have no Idea about the impact of porting our current build script to FAKE5. From the 600+ lines I guess we may run into a few issues.

My current workaround: Install VS2017 Build Tools incl. all target frameworks, .net desktop and web stuff.

**Comment by BlythMeister** [comment]
I thought that change was already released...

**Comment by BlythMeister** [comment]
The most recent FAKE4 is 4.64.17
You just need to update and it will work.

Check nuget https://www.nuget.org/packages/FAKE/4.64.17

**Comment by embix** [comment]
@BlythMeister cannot update FAKE atm due to timeouts from nuget.org. Will try that again later.

Any reason why the 4.64.16 and 4.64.17 tags weren't pushed?

**Comment by matthid** [comment]
> Any reason why the 4.64.16 and 4.64.17 tags weren't pushed?

Probably because this process isn't as automated as it was historically. This is because we don't want to publish docs when releasing fake 4 for example. Releasing older versions is something we have never done in the past ;)

**Comment by embix** [comment]
I see. It would certainly help to know, which commit was used for the release. Even the (short) commit hash as part of file/product version would suffice.
Anyways, thanks for the support 👍

**Comment by matthid** [comment]
>  I have no Idea about the impact of porting our current build script to FAKE5. From the 600+ lines I guess we may run into a few issues.

Just one word regarding this: It shouldn't break. But it will add lots of obsolete warnings and it will hint you that you are using the "legacy" runner. Anyway, I guess this can be closed :)
